dev/README.md

```diff
@@ -0,0 +1,559 @@
+# Developing Mountpoint for Amazon S3 CSI Driver
+
+This document and the `dev/` folder contains references, documentation, and various things for developing the CSI Driver. Note that this document is written with v2 in mind, most things should also apply to v1 but there might be slight differences.
+
+## Background
+
+The CSI Driver implements [Container Storage Interface (CSI)](https://github.com/container-storage-interface/spec/blob/v1.9.0/spec.md) to provide Amazon S3 Buckets as volumes for containers using [Mountpoint for Amazon S3](https://github.com/awslabs/mountpoint-s3) under the hood.
+Mountpoint is based on Filesystem in Userspace (FUSE) and allows you to mount your buckets as local file systems.
+
+Even though the CSI is a generic interface, Mountpoint for Amazon S3 CSI Driver actively only test against [Kubernetes](https://kubernetes.io/).
+All end-to-end tests of the CSI Driver creates [Amazon Elastic Kubernetes Service (EKS)](https://aws.amazon.com/eks/) clusters,
+installs the CSI Driver, and runs the test against the real EKS cluster.
+
+### Kubernetes primer
+
+Kubernetes is a container orchestration system with the following architecture:
+
+![Kubernetes Architecture](https://kubernetes.io/images/docs/components-of-kubernetes.svg)
```

Comment:
np: This and next diagram looks bad in dark mode 😅

<img width="1066" height="500" alt="Screenshot 2025-08-22 at 13 22 15" src="https://github.com/user-attachments/assets/13883d02-aeaf-4441-b668-16f4efe21605" />

Reply:
Should be better now